src/cosalib/fedora_messaging_request.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/python3
+
+'''
+    Implements signing with RoboSignatory via fedora-messaging. To run this,
+    one needs credentials to the restricted Fedora broker. In a developer
+    workflow, one can also run it (and RoboSignatory) against a local rabbitmq
+    instance. For more details, see:
+
+    https://fedora-messaging.readthedocs.io/en/latest/quick-start.html
+'''
+
+import copy
+import threading
+import uuid
+
+import multiprocessing as mp
+
+from fedora_messaging import message
+from fedora_messaging.api import publish, twisted_consume
+from fedora_messaging.config import conf
+
+from twisted.internet import reactor
+
+# these files are part of fedora-messaging
+FEDORA_MESSAGING_PUBLIC_CONF = {
+    'prod': '/etc/fedora-messaging/fedora.toml',
+    'stg': '/etc/fedora-messaging/fedora.stg.toml',
+}
+# https://apps.fedoraproject.org/datagrepper/raw?topic=org.fedoraproject.prod.coreos.build.request.ostree-sign&delta=100000
+# https://apps.fedoraproject.org/datagrepper/raw?topic=org.fedoraproject.prod.coreos.build.request.artifacts-sign&delta=100000
+FEDORA_MESSAGING_TOPIC_PREFIX = {
+    'prod': 'org.fedoraproject.prod.coreos.build.request',
+    'stg': 'org.fedoraproject.stg.coreos.build.request',
+}
+
+# Default to timeout after 60 seconds
+DEFAULT_REQUEST_TIMEOUT_SEC = 60
+
+def send_request_and_wait_for_response(config=None,
+                                       environment='prod',
+                                       request_type=None,
```

Comment:
Feels like this one shouldn't have a default. E.g. maybe let's change the signature to `..._response(request_type, config=None, ...)`?

Reply:
should be fixed in the next upload